addons/hr_expense/models/hr_expense.py

```diff
@@ -863,6 +871,15 @@ def _compute_amount(self):
         for sheet in self:
             sheet.total_amount = sum(sheet.expense_line_ids.mapped('total_amount_company'))
 
+    @api.depends('account_move_id.payment_state')
+    def _compute_payment_state(self):
+        for sheet in self:
+            move = sheet.account_move_id
+            if move and move.payment_state:
+                sheet.payment_state = move.payment_state if move.payment_state != 'invoicing_legacy' else False
```

Comment:
To avoid unconsistent state in case of new value added to this selection field:
`[opt[0] for opt in self._fields['payment_state']['selection'] if opt[0] == move.payment_state] or False`
Are we sure about `False`?
Maybe we should add the selection inside a class variable and import it into expense?

Reply:
not getting your point: how does that help when we add new selection to payment_state in account.move, or in here?

Reply:
To ensure both fields are still consistent with each others. I suggest
```
PAYMENT_STATE_SELECTION = [
        ('not_paid', 'Not Paid'),
        ('in_payment', 'In Payment'),
        ('paid', 'Paid'),
        ('partial', 'Partially Paid'),
        ('reversed', 'Reversed'),
        ('invoicing_legacy', 'Invoicing App Legacy'),
]
```
... to be used in both fields. Then, `sheet.payment_state = move.payment_state`.

Reply:
Note: If `payment_state` is `False`, then
```
                    <widget name="web_ribbon" title="Paid" bg_color="bg-success" attrs="{'invisible': [('payment_state', '!=', 'paid')]}"/>
                    <widget name="web_ribbon" title="Partial" bg_color="bg-info" attrs="{'invisible': [('payment_state', '!=', 'partial')]}"/>
                    <widget name="web_ribbon" title="Reversed" bg_color="bg-danger" attrs="{'invisible': [('payment_state', '!=', 'reversed')]}"/>
                    <widget name="web_ribbon" title="In Payment" attrs="{'invisible': [('payment_state', '!=', 'in_payment')]}"/>
```
is displaying all the ribbons

Reply:
`False != 'paid'` ==> invisible
None is showed

Reply:
@william-andre Indeed^^ My bad.
@jbw-odoo I think we could rewrite 
```
for sheet in self:
    move = sheet.account_move_id
    if move and move.payment_state:
         sheet.payment_state = move.payment_state if move.payment_state != 'invoicing_legacy' else False
    else:
         sheet.payment_state = 'not_paid'
```
to
```
for sheet in self:
     sheet.payment_state = sheet.account_move_id.payment_state
```
What do you think?

Reply:
Then no status is shown when no move. 
![image](https://user-images.githubusercontent.com/56965943/109484482-c980dc00-7a80-11eb-8443-35c115b5fe1a.png)

Reply:
`        for sheet in self:
            sheet.payment_state = sheet.account_move_id.payment_state or 'not_paid'`
?

Reply:
@jbw-odoo ok for the `or` :)

Reply:
pushed